mmcls/apis/inference.py

```diff
@@ -55,66 +56,34 @@ def inference_model(model, img):
     """Inference image(s) with the classifier.
 
     Args:
-        model (nn.Module): The loaded classifier.
+        model (BaseClassifier): The loaded classifier.
         img (str/ndarray): The image filename or loaded image.
 
     Returns:
         result (dict): The classification results that contains
             `class_name`, `pred_label` and `pred_score`.
     """
     cfg = model.cfg
-    device = next(model.parameters()).device  # model device
     # build the data pipeline
+    test_pipeline_cfg = cfg.test_dataloader.dataset.pipeline
     if isinstance(img, str):
-        if cfg.data.test.pipeline[0]['type'] != 'LoadImageFromFile':
-            cfg.data.test.pipeline.insert(0, dict(type='LoadImageFromFile'))
-        data = dict(img_info=dict(filename=img), img_prefix=None)
+        if test_pipeline_cfg[0]['type'] != 'LoadImageFromFile':
+            test_pipeline_cfg.insert(0, dict(type='LoadImageFromFile'))
+        data = dict(img_path=img)
     else:
-        if cfg.data.test.pipeline[0]['type'] == 'LoadImageFromFile':
-            cfg.data.test.pipeline.pop(0)
+        if test_pipeline_cfg[0]['type'] == 'LoadImageFromFile':
+            test_pipeline_cfg.pop(0)
         data = dict(img=img)
-    test_pipeline = Compose(cfg.data.test.pipeline)
+    test_pipeline = Compose(test_pipeline_cfg)
     data = test_pipeline(data)
-    data = collate([data], samples_per_gpu=1)
-    if next(model.parameters()).is_cuda:
-        # scatter to specified GPU
-        data = scatter(data, [device])[0]
+    data = pseudo_collate([data])
 
     # forward the model
     with torch.no_grad():
-        scores = model(return_loss=False, **data)
-        pred_score = np.max(scores, axis=1)[0]
-        pred_label = np.argmax(scores, axis=1)[0]
+        prediction = model.val_step(data)[0].pred_label
+        pred_score = torch.max(prediction.score).item()
+        pred_label = torch.argmax(prediction.score).item()
```

Comment:
This does work in multi-label classification tasks, why not just return a pred label?

Reply:
This is just a temporary fix to ensure the original result.